lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -2037,7 +2037,8 @@ class ModuleWriter {
   }
 
   void writePrologue(raw_ostream &out) {
-    out << "// Generated by " << version::getSwiftFullVersion() << "\n"
+    out << "// Generated by " << version::getSwiftFullVersion(
+      M.getASTContext().LangOpts.EffectiveLanguageVersion) << "\n"
```

Comment:
Using `getFullSwiftVersion()` is reasonable. I was convincing myself that we wanted '3.0' in there when emulating 3.0, but that seems like the right idea.